src/azul/service/manifest_service.py

```diff
@@ -701,13 +708,8 @@ def _option(cls, s: str):
         return '"' + s.replace('\\', '\\\\').replace('"', '\\"') + '"'
 
     def write_to(self, output: IO[str]) -> Optional[str]:
-        output.write('--create-dirs\n\n'
-                     '--compressed\n\n'
-                     '--location\n\n'
-                     '--continue-at -\n\n')
-        for hit in self._create_request().scan():
-            doc = self._hit_to_doc(hit)
-            file = one(doc['contents']['files'])
+
+        def _write(file: dict):
```

Comment:
This suggestion is very optional, because the diff is very small, but you're combining refactoring (creating the `_write()` function) with functional changes. It would be easier to understand the diff if you moved the refactoring into a separate, prior commit.

Reply:
I'm actually having a harder time understanding this if the history is split up like it is now. The refactoring doesn't make much sense in isolation. Why would I extract a function with just one call site? If we roll back the functional change, we'd also want to roll back the refactoring.

Reply:
For larger changes, @jessebrennan's suggestion would make sense, especially if the commits are pushed separately. It's just that this change is so small.